### DISCUSSION TITLE
W8912: Incompatible with Unity Catalog: from boto3 import Session (incompatible-with-uc)

### DISCUSSION THREAD

**Discussion by NodeJSmith** [databrickslabs/pylint-plugin#36] [Q&A] [answered]
Is there a way to customize what is marked as incompatible with UC and/or a way to mark a line to be skipped during the check?

One of our tasks imports boto3 to remove files from S3, but it uses secrets set on the cluster to create the session, so it's totally compatible. Come to think of it, I'm not actually sure what legacy behavior this is checking for, was boto3 used for hive metastore operations?

PS: Love the idea of this project, I was recently bit several times by inadvertent global spark variables and have been checking things manually since then, but I'll be adding this to my CI pipelines for this going forward.

**Comment by nfx** [comment, marked as answer]
Our current detection is a bit more aggressive than it should be. But it's kind of straightforward at the moment: see https://github.com/databrickslabs/pylint-plugin/blob/main/src/databricks/labs/pylint/legacy.py

There are two options: 

- add "#pylint: disable=check-name"
- make a PR to take boto3 out of generic list and add logic for all downstream calls to be checked against their initialization
